storage/blockchain/src/ops/output.rs

```diff
@@ -155,9 +158,31 @@ pub fn output_to_output_on_chain(
     amount: Amount,
     table_tx_unlock_time: &impl DatabaseRo<TxUnlockTime>,
 ) -> Result<OutputOnChain, RuntimeError> {
-    // FIXME: implement lookup table for common values:
-    // <https://github.com/monero-project/monero/blob/c8214782fb2a769c57382a999eaf099691c836e7/src/ringct/rctOps.cpp#L322>
-    let commitment = ED25519_BASEPOINT_POINT + *H * Scalar::from(amount);
+    // Binary search in lookup table (O(log2(n)+2) instead of O(n+2))
+    let commitment: EdwardsPoint = {
+        let (mut start, mut end, mut c): (usize, usize, Option<EdwardsPoint>) =
+            (0, (ZC_LT.len() - 1) / 2, None);
+        while start <= end {
+            let mid = (start + end) / 2;
+            let lookup = ZC_LT[mid].0;
+            match amount {
+                a if a == lookup => {
+                    c = CompressedEdwardsY::from_slice(&ZC_LT[mid].1)
+                        .expect("The lookup table cannot contain invalid curve point data")
+                        .decompress();
```

Comment:
We could keep the decomposed point in the cache instead of doing a decompression on every use.